libvast/src/arrow_table_slice.cpp

```diff
@@ -1251,6 +1251,10 @@ upgrade_array_to_v2(const std::shared_ptr<arrow::Array>& arr, const type& t) {
       VAST_ASSERT(arr->type()->Equals(t.to_arrow_type()));
       return arr;
     },
+    [&](const duration_type&) -> std::shared_ptr<arrow::Array> {
+      // duration is backed by physical int64 array, but logical type differs
+      return arr;
```

Comment:
You're deep in undefined behavior land here with pointer aliasing rules. I think the correct way to handle this is to create a new Array from the existing Array's underlying ArrayData.

```suggestion
      // Duration is backed by physical Int64 array, but logical type differs.
      auto result = arr->View(duration_type::to_arrow_type());
      VAST_ASSERT(result.ok(), result.status().ToString().c_str());
      return result.MoveValueUnsafe();
```